interfaces/builtin/download.go

```diff
@@ -0,0 +1,325 @@
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=defaultThrottle
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=isGSMDownloadAllowed
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Explicitly deny DownloadManager APIs apps shouldn't have access to in order
+# to make sure they aren't accidentally added in the future (see LP: #1277578
+# for details)
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=allowGSMDownload,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=createMmsDownload,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=exit,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=setDefaultThrottle,
+`)
+
+var downloadPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as a download manager.
+# Usage: common
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+#include <abstractions/dbus-session-strict>
+
+# https://specifications.freedesktop.org/download-spec/latest/
+# allow binding to the DBus download interface
+dbus (bind)
+    bus=session
+    name="com.canonical.applications.Downloader",
+
+# Allow writing to app download directories
+owner @{HOME}/snap/*/common/Downloads/    rw,
+owner @{HOME}/snap/*/common/Downloads/**  rwk,
+
+dbus (send)
+    bus={system,session}
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus={system,session}
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{RequestName,ReleaseName}"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/
+    interface=org.freedesktop.DBus
+    member="GetConnectionAppArmorSecurityContext"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=org.freedesktop.DBus
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=org.freedesktop.DBus.Properties
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+     bus=session
+     path=/com/canonical/applications/download/**
+     interface=com.canonical.applications.Download
+     peer=(label=unconfined),
+
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=downloadCreated
+     peer=(label=unconfined),
+`)
```

Comment:
I expected these to use `label=###PLUG_SECURITY_TAGS###` but you have `label=unconfined`. Can you explain? Is this for talking to applications shipped as debs on classic?